sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -332,8 +332,12 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
         execution.Sample(lb, ub, withReplacement, seed, planLater(child)) :: Nil
       case logical.LocalRelation(output, data) =>
         LocalTableScan(output, data) :: Nil
-      case logical.Limit(IntegerLiteral(limit), child) =>
-        execution.Limit(limit, planLater(child)) :: Nil
+      case logical.Limit(IntegerLiteral(limit), child) => {
+        val perPartitionLimit = execution.PartitionLocalLimit(limit, planLater(child))
+        val globalLimit = execution.PartitionLocalLimit(
+          limit, execution.Exchange(SinglePartition, Nil, perPartitionLimit))
```

Comment:
_Actually_, I take _that_ back: maybe `global` is the right name for the parameter, similar to Sort.